core/src/main/java/org/primefaces/extensions/component/inputpin/InputPinRenderer.java

```diff
@@ -0,0 +1,164 @@
+        final InputPin inputPin = (InputPin) component;
+
+        if (!shouldDecode(inputPin)) {
+            return;
+        }
+
+        final String inputId = inputPin.getClientId(context) + HIDDEN_ID;
+        final String submittedValue = context.getExternalContext().getRequestParameterMap().get(inputId);
+
+        if (submittedValue != null) {
+            inputPin.setSubmittedValue(submittedValue);
+        }
+
+        decodeBehaviors(context, inputPin);
+    }
+
+    @Override
+    public void encodeEnd(final FacesContext context, final UIComponent component) throws IOException {
+        final InputPin inputPin = (InputPin) component;
+
+        if (inputPin.getSize() < 1) {
+            throw new FacesException("InputPin size property is required.");
+        }
+
+        String valueToRender = ComponentUtils.getValueToRender(context, inputPin, inputPin.getValue());
+        if (valueToRender == null) {
+            valueToRender = Constants.EMPTY_STRING;
+        }
+
+        encodeMarkup(context, inputPin, valueToRender);
+        encodeScript(context, inputPin);
+    }
+
+    protected void encodeMarkup(final FacesContext context, final InputPin inputPin, final String valueToRender)
+                throws IOException {
+        final ResponseWriter writer = context.getResponseWriter();
+        final String clientId = inputPin.getClientId(context);
+        final String styleClass = getStyleClassBuilder(context)
+                    .add(InputPin.STYLE_CLASS, inputPin.getStyleClass())
+                    .build();
+
+        writer.startElement("span", inputPin);
+        writer.writeAttribute("id", clientId, null);
+        writer.writeAttribute(Attrs.CLASS, styleClass, "styleClass");
+
+        if (inputPin.getStyle() != null) {
+            writer.writeAttribute(Attrs.STYLE, inputPin.getStyle(), Attrs.STYLE);
+        }
+
+        encodeInput(context, inputPin, clientId, valueToRender);
+        encodeHiddenInput(context, inputPin, clientId, valueToRender);
+
+        writer.endElement("span");
+    }
+
+    protected void encodeInput(final FacesContext context, final InputPin inputPin, final String clientId,
+                final String valueToRender)
+                throws IOException {
+
+        final ResponseWriter writer = context.getResponseWriter();
+        final String inputStyle = inputPin.getInputStyle();
+        final String inputStyleClass = getStyleClassBuilder(context)
+                    .add(InputText.STYLE_CLASS, InputPin.CELL_STYLE_CLASS)
+                    .add(inputPin.getInputStyleClass())
+                    .build();
+        final char[] chars = valueToRender.toCharArray();
+        for (int i = 1; i <= inputPin.getSize(); i++) {
+            final String inputId = clientId + InputPin.INPUT_SUFFIX + i;
+            final String inputValue = chars.length >= i ? String.valueOf(chars[i - 1]) : "";
+
+            writer.startElement("input", null);
```

Comment:
Do we need maxlength 1 here?

Reply:
with maxlength="1" it wouldn't be possible to paste longer text (AFAIK)